src/System.Slices/System/Buffers/PinnedMemoryHandle.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    public unsafe struct PinnedMemoryHandle<T>
```

Comment:
Only if you create it with GCHandleType.Pinned. Other GCHandleTypes has nothing to do with pinning.

I am fine with this being called MemoryHandle ... I was just pointing out that the pinning is not implied in GCHandle.